src/test/java/com/fasterxml/jackson/databind/ser/JsonPropertyOrderWithAnyGetter4388Test.java

```diff
@@ -0,0 +1,304 @@
+                "'entityId':1," +
+                "'totalTests':2," +
+                "'child1':3," +
+                "'child2':3," +
+                "'entityName':'Bob'," +
+                "'product1':4}"),
+            json);
+    }
+
+    @Test
+    public void testIgnoreProperties() throws Exception {
+        // Respsect @JsonIgnoreProperties 'b' from Pojo, but not from map
+        IgnorePropertiesOnFieldPojo bean = new IgnorePropertiesOnFieldPojo();
+        bean.map.put("b", 3);
+        assertEquals(a2q("{'a':1,'b':3}"), MAPPER.writeValueAsString(bean));
+
+        // Respect @JsonIgnoreProperties 'b' from Pojo, but not from map
+        IgnorePropertiesOnAnyGetterPojo bean2 = new IgnorePropertiesOnAnyGetterPojo();
+        bean2.map.put("b", 3);
+        assertEquals(a2q("{'a':1,'b':2}"), MAPPER.writeValueAsString(bean2));
+
+        // Respect @JsonIgnore from Pojo, but not from map
+        IgnoreOnFieldPojo bean3 = new IgnoreOnFieldPojo();
+        bean3.map.put("b", 3);
+        assertEquals(a2q("{'a':1,'b':3}"), MAPPER.writeValueAsString(bean3));
+    }
+
+    // Sorting works on @JsonAnyGetter, when adding @JsonPropertyOrder directly on the AnyGetter method
+    @Test
+    public void testSortingOnAnyGetter() throws Exception {
+        AlphabeticOrderOnAnyGetterBean bean = new AlphabeticOrderOnAnyGetterBean();
+        bean.map.put("zd", 4);
+        bean.map.put("zc", 3);
+        bean.map.put("za", 1);
+        bean.map.put("zb", 2);
+        assertEquals(a2q("{" +
+            "'za':1," +
+            "'zb':2," +
+            "'zc':3," +
+            "'zd':4}"), MAPPER.writeValueAsString(bean));
+    }
+
+    // Sorting does not work on @JsonAnyGetter, when adding @JsonPropertyOrder on the class
+    @Test
+    public void testSortingOnClassNotPropagateToAnyGetter() throws Exception {
+        AlphabeticOrderOnClassBean bean = new AlphabeticOrderOnClassBean();
+        bean.map.put("zc", 3);
+        bean.map.put("za", 1);
+        bean.map.put("zb", 2);
+        assertEquals(a2q("{" +
+            "'a':1," +
+            "'b':2," +
+            "'c':3," +
+            "'zc':3," +
+            "'za':1," +
+            "'zb':2}"), MAPPER.writeValueAsString(bean));
+    }
+
+    @Test
+    public void testLinkUnlinkWithJsonIgnore() throws Exception {
+        LinkUnlinkConflictPojo pojo = new LinkUnlinkConflictPojo();
+        String json = MAPPER.writeValueAsString(pojo);
+
+        assertEquals(a2q("{'key':'value'}"), json);
+    }
+
+
+    @Test
+    public void testPrivateAnyGetter() throws Exception {
+        PrivateAnyGetterPojo pojo = new PrivateAnyGetterPojo();
+        pojo.add("secondProperty", 2);
+        String json = MAPPER.writeValueAsString(pojo);
+
+        assertEquals(a2q("{" +
+                "'firstProperty':1," +
+                "'thirdProperty':3," +
+                "'forthProperty':4," +
+                "'secondProperty':2}"), // private accesor, wont' work here
+            json);
+    }
```

Comment:
Here we have test case where any-getter is private without proper getter to be detected for sorting, so  `"secondProperty"` field will be appended in the end.